cmd/uplink/cmd/cp.go

```diff
@@ -50,44 +49,67 @@ func upload(ctx context.Context, bs buckets.Store, src fpath.FPath, dst fpath.FP
 		dst = dst.Join(src.Base())
 	}
 
-	var f *os.File
+	var file *os.File
 	var err error
 	if src.Base() == "-" {
-		f = os.Stdin
+		file = os.Stdin
 	} else {
-		f, err = os.Open(src.Path())
+		file, err = os.Open(src.Path())
 		if err != nil {
 			return err
 		}
-		defer utils.LogClose(f)
+		defer utils.LogClose(file)
 	}
 
-	fi, err := f.Stat()
+	fileInfo, err := file.Stat()
 	if err != nil {
 		return err
 	}
 
-	if fi.IsDir() {
+	if fileInfo.IsDir() {
 		return fmt.Errorf("source cannot be a directory: %s", src)
 	}
 
-	o, err := bs.GetObjectStore(ctx, dst.Bucket())
+	metainfo, streams, err := cfg.Metainfo(ctx)
+	if err != nil {
+		return err
+	}
+
+	bucket, err := metainfo.GetBucket(ctx, dst.Bucket())
```

Comment:
I think you should be able to use `metainfo.CreateObject` directly.

Reply:
Wait, didn't we add the "Bucket" info to mutable object or mutable stream?

Reply:
We have just discussed it, but nothing done yet about this.

Reply:
ah, ok, let's leave as is then